### PR TITLE
Typechecking Pride: add a few files to pride list, and fix extern cal…

### PR DESCRIPTION
**Pull request by samouri**
Split off partial of: https://github.com/ampproject/amphtml/pull/33957

- fixes extern calculation
- Comments out parts of `amp.extern.js` that would fail without including src files
- Fixes typing in dompurify extern
- Moves a json extern that is only used within `src/json.js` to from amp.extern.js to the json file.

### CODE REVIEW

File: build-system/tasks/check-types.js [hunk 1]
```diff
@@ -35,7 +36,22 @@ const {typecheckNewServer} = require('../server/typescript-compile');
  * Note: This is a TEMPORARY holding point during the transition to type-safety.
  * @type {!Array<string>}
  */
-const PRIDE_FILES_GLOBS = ['src/resolved-promise.js', 'src/types.js'];
+const PRIDE_FILES_GLOBS = [
+  // Core
+  'src/core{,/**}/*.js',
```
Comment (rcebulko) [inline build-system/tasks/check-types.js:41]:
```suggestion
  'src/core/**/*.js',
```

File: build-system/externs/amp.extern.js [hunk 2]
```diff
@@ -80,6 +80,25 @@ Element.prototype.__AMP_SHADOW_ROOT;
 /** @type {?ShadowRoot} */
 Element.prototype.shadowRoot;
 
+// Fullscreen methods
+Document.prototype.cancelFullScreen = function () {};
```
Comment (rcebulko) [inline build-system/externs/amp.extern.js:84]: Do we need to assign these function values here?

File: build-system/externs/amp.extern.js [hunk 3]
```diff
@@ -241,11 +270,20 @@ window.AMP.toggleExperiment;
 window.AMP.setLogLevel;
 window.AMP.setTickFunction;
 window.AMP.viewer;
-window.AMP.viewport = {};
+window.AMP.viewport;
```
Comment (rcebulko) [inline build-system/externs/amp.extern.js:273]: Are these statements declaring everything here as `?Object`?

File: build-system/externs/amp.extern.js [hunk 4]
```diff
@@ -476,9 +509,15 @@ AmpElement.prototype.togglePlaceholder = function (show) {};
 /** @return {{width: number, height: number}} */
 AmpElement.prototype.getLayoutSize = function () {};
 
+/**
+ * TODO: remove this when typechecking is restored to AMP.BaseElement.
+ * @typedef {*}
+ */
+var BaseElement;
```
Comment (rcebulko) [inline build-system/externs/amp.extern.js:516]: We should probably try standardize on `let`, at least for new entries

File: build-system/externs/amp.extern.js [hunk 5]
```diff
@@ -752,49 +791,50 @@ AmpStoryVariableService.prototype.onStateChange = function (event) {};
...
- * @constructor @struct
- * @extends {AmpAdUIHandler$$module$extensions$amp_ad$0_1$amp_ad_ui}
- */
-AMP.AmpAdUIHandler = class {
-  /**
-   * @param {!AMP.BaseElement} baseInstance
-   */
-  constructor(baseInstance) {}
-};
+// TODO: uncomment when typechecking restored to BaseElement.
```
Comment (rcebulko) [inline build-system/externs/amp.extern.js:794]: Do we have a tracking issue for this work yet?

Comment (rcebulko) [reply build-system/externs/amp.extern.js:794]: Generated this from our doc https://github.com/ampproject/amphtml/issues/34096

Comment (samouri) [reply build-system/externs/amp.extern.js:794]: No, but I'm hoping to iterate quickly here.

Comment (samouri) [reply build-system/externs/amp.extern.js:794]: https://github.com/ampproject/amphtml/issues/34099

File: build-system/externs/amp.extern.js (hunk 3 above)
Comment (samouri) [reply build-system/externs/amp.extern.js:273]: I have no idea what this defaults to, I've been assuming `?`

File: build-system/externs/amp.extern.js (hunk 2 above)
Comment (samouri) [reply build-system/externs/amp.extern.js:84]: no we don't, although I figure it is slightly better than `?`

File: build-system/tasks/check-types.js (hunk 1 above)
Comment (samouri) [reply build-system/tasks/check-types.js]: 🙃

File: build-system/externs/amp.extern.js (hunk 4 above)
Comment (samouri) [reply build-system/externs/amp.extern.js:516]: Updating all to `var`

File: build-system/externs/amp.extern.js (hunk 2 above)
Comment (samouri) [reply build-system/externs/amp.extern.js:84]: I removed the function

**Review by rcebulko — APPROVED** [review]

File: build-system/tasks/check-types.js [hunk 6]
```diff
@@ -188,9 +205,11 @@ async function typeCheck(targetName) {
 
   // If srcGlobs and externGlobs are defined, determine the externs/extraGlobs
   if (srcGlobs.length || externGlobs.length) {
-    opts.eterns = externGlobs.flatMap(globby.sync);
+    opts.externs = externGlobs.flatMap(globby.sync);
```
Comment (rsimha) [inline build-system/tasks/check-types.js:208]: _Eternal joy_ or _external joy_, that is the question 😃

File: build-system/tasks/check-types.js [hunk 7]
```diff
@@ -233,7 +252,7 @@ async function checkTypes() {
   cleanupBuildDir();
   maybeInitializeExtensions();
   typecheckNewServer();
-  await compileCss();
+  await Promise.all([compileCss(), compileJison()]);
```
Comment (rsimha) [inline build-system/tasks/check-types.js:255]: [This](https://docs.google.com/document/d/1vac_qbUGQ_eQiEIwhVQB5G4ASYtI509ZRhIZklJj7zw/edit) proposal by @alanorozco may make this unnecessary in future. 🤞

File: build-system/test-configs/forbidden-terms.js [hunk 8]
```diff
@@ -1061,7 +1061,11 @@ const forbiddenTermsSrcInclusive = {
   '\\.indexOf\\(.*===?.*\\.length': 'use endsWith helper in src/string.js',
   '/url-parse-query-string': {
     message: 'Import parseQueryString from `src/url.js`',
-    allowlist: ['src/url.js', 'src/mode.js'],
+    allowlist: [
+      'build-system/tasks/check-types.js',
```
Comment (rsimha) [inline build-system/test-configs/forbidden-terms.js:1065]: Are you still using `indexOf` in `check-types.js`? If it's a nodejs string operation, I'd imagine you can use `.includes`, `.startsWith`, etc.?

**Review by rsimha — APPROVED** [review]
Changes to `check-types.js` LGTM with one nit to verify.

File: build-system/tasks/check-types.js (hunk 7 above)
Comment (samouri) [reply build-system/tasks/check-types.js:255]: 🤞

File: build-system/test-configs/forbidden-terms.js (hunk 8 above)
Comment (samouri) [reply build-system/test-configs/forbidden-terms.js:1065]: I'm so confused...how does this question relate to this PR?

The added line you've commented on is due to my typechecking a filename that is "forbidden" to use.

File: build-system/tasks/check-types.js (hunk 6 above)
Comment (samouri) [reply build-system/tasks/check-types.js:208]: 🤣

File: build-system/tasks/check-types.js (hunk 7 above)
Comment (alanorozco) [reply build-system/tasks/check-types.js:255]: 🤞

File: build-system/test-configs/forbidden-terms.js (hunk 8 above)
Comment (rsimha) [reply build-system/test-configs/forbidden-terms.js:1065]: Crap, I misread this. (I thought this had to do with `indexOf` on line 1061, but it's really about `url-parse-query-string` on line 1062. Carry on.

Comment (rcebulko) [reply build-system/test-configs/forbidden-terms.js:1065]: I was also very confused 😆

**Pull request merged**